samples/nrf9160/secure_boot/src/main.c

```diff
@@ -65,166 +65,195 @@
+	  << SPU_FLASHREGION_PERM_SECATTR_Pos) &                               \
+	 SPU_FLASHREGION_PERM_SECATTR_Msk)
+
+#define SRAM_EXEC                                                              \
+	((SPU_RAMREGION_PERM_EXECUTE_Enable                                    \
+	  << SPU_RAMREGION_PERM_EXECUTE_Pos) &                                 \
+	 SPU_RAMREGION_PERM_EXECUTE_Msk)
+
+#define SRAM_WRITE                                                             \
+	((SPU_RAMREGION_PERM_WRITE_Enable << SPU_RAMREGION_PERM_WRITE_Pos) &   \
+	 SPU_RAMREGION_PERM_WRITE_Msk)
+
+#define SRAM_READ                                                              \
+	((SPU_RAMREGION_PERM_READ_Enable << SPU_RAMREGION_PERM_READ_Pos) &     \
+	 SPU_RAMREGION_PERM_READ_Msk)
+
+#define SRAM_LOCK                                                              \
+	((SPU_RAMREGION_PERM_LOCK_Locked << SPU_RAMREGION_PERM_LOCK_Pos) &     \
+	 SPU_RAMREGION_PERM_LOCK_Msk)
+
+#define SRAM_SECURE                                                            \
+	((SPU_RAMREGION_PERM_SECATTR_Secure                                    \
+	  << SPU_RAMREGION_PERM_SECATTR_Pos) &                                 \
+	 SPU_RAMREGION_PERM_SECATTR_Msk)
+
+#define SRAM_NONSEC                                                            \
+	((SPU_RAMREGION_PERM_SECATTR_Non_Secure                                \
+	  << SPU_RAMREGION_PERM_SECATTR_Pos) &                                 \
+	 SPU_RAMREGION_PERM_SECATTR_Msk)
+
+#define PERIPH_PRESENT                                                         \
+	((SPU_PERIPHID_PERM_PRESENT_IsPresent                                  \
+	  << SPU_PERIPHID_PERM_PRESENT_Pos) &                                  \
+	 SPU_PERIPHID_PERM_PRESENT_Msk)
+
+#define PERIPH_NONSEC                                                          \
+	((SPU_PERIPHID_PERM_SECATTR_NonSecure                                  \
+	  << SPU_PERIPHID_PERM_SECATTR_Pos) &                                  \
+	 SPU_PERIPHID_PERM_SECATTR_Msk)
+
+#define PERIPH_DMA_NOSEP                                                       \
+	((SPU_PERIPHID_PERM_DMA_NoSeparateAttribute                            \
+	  << SPU_PERIPHID_PERM_DMA_Pos) &                                      \
+	 SPU_PERIPHID_PERM_DMA_Msk)
+
+#define PERIPH_LOCK                                                            \
+	((SPU_PERIPHID_PERM_LOCK_Locked << SPU_PERIPHID_PERM_LOCK_Pos) &       \
+	 SPU_PERIPHID_PERM_LOCK_Msk)
+
 static void secure_boot_config_flash(void)
 {
-	int i;
-
 	/* Lower 256 kB of flash is allocated to MCUboot (if present)
 	 * and to the Secure firmware image. The rest of flash is
-	 * allocated to Non-Secure firmware image, thus SPU attributes
-	 * it as Non-Secure.
+	 * allocated to Non-Secure firmware image.
 	 */
-	printk("Secure Boot: configure flash\n");
+	static const u32_t flash_perm[] = {
+		/* Configuration for Regions 0 - 7 (0 - 256 kB) */
+		[0 ... 7] = FLASH_READ | FLASH_WRITE | FLASH_EXEC |
```

Comment:
How does this check for the presences of MCUBoot? I know it's an RFC so I guess it is not? Anyways shouldn't it be configurable the amount of flash you want to lock?

Reply:
Are the segments always contiguous? If not it might be too many options.

Reply:
Any bootloader stage prior to the stage configuring the SPU will have set their own configurations, and need not be considered. 

AFAIK, the bootloader stage which configures the SPU will also contain any other secure services (since the locking granularity is 32kB, we need to place something there). So typically the secure areas will all be located before the non-secure areas.

IMO it is safe to assume that all secure flash is placed before non secure flash.